portal/portal_payment.php

```diff
@@ -1269,81 +1349,311 @@ class="control-label"><?php echo xlt('Card Expiry Date'); ?></label>
+                                    </div>
+                                </div>
+                                <div class="form-row form-group">
+                                    <label for="card-element"><?php echo xlt('Credit or Debit Card') ?></label>
+                                    <div id="card-element"></div>
+                                    <div id="card-errors" role="alert"></div>
+                                </div>
+                                <div class="col-md-6">
+                                    <h4 style="display: inline-block;"><?php echo xlt('Payment Amount'); ?>:&nbsp;
+                                        <strong><span id="payTotal"></span></strong></h4>
+                                </div>
+                                <input type='hidden' name='mode' id='mode' value=''/>
+                                <input type='hidden' name='cc_type' id='cc_type' value=''/>
+                                <input type='hidden' name='payment' id='paymentAmount' value=''/>
+                                <input type='hidden' name='invValues' id='invValues' value=''/>
+                            </fieldset>
+                        </form>
+                    <?php } ?>
                 </div>
                 <!-- Body  -->
                 <div class="modal-footer">
-                    <button type="button" class="btn btn-default"
-                            data-dismiss="modal"><?php echo xlt('Close'); ?></button>
+                    <div class="button-group">
+                        <button type="button" class="btn btn-default" data-dismiss="modal"><?php echo xlt('Cancel'); ?></button>
+                        <?php if ($GLOBALS['payment_gateway'] == 'InHouse') { ?>
+                            <button id="paySubmit" class="btn btn-primary"><?php echo xlt('Send Payment'); ?></button>
+                        <?php } else if ($GLOBALS['payment_gateway'] == 'AuthorizeNet') { ?>
+                            <button id="payAurhorizeNet" class="btn btn-primary" onclick="sendPaymentDataToAnet(event)"><?php echo xlt('Pay Now'); ?></button>
+                        <?php }
+if ($GLOBALS['payment_gateway'] == 'Stripe') { ?>
+                            <button id="stripeSubmit" class="btn btn-primary"><?php echo xlt('Pay Now'); ?></button>
+                        <?php } ?>
+                    </div>
                 </div>
             </div>
         </div>
     </div>
-    <?php echo "<script>var ccerr='" . xlt('Invalid Credit Card Number') . "';</script>"; ?>
     <script type="text/javascript">
-        $('#cc_number').validateCreditCard(function (result) {
-            var r = (result.card_type == null ? '' : result.card_type.name.toUpperCase())
-            var v = (result.valid == true ? ' Valid #' : ' Invalid #')
-            $('#cardtype').val(r + v);
+        var ccerr = <?php echo json_encode(xl('Invalid Credit Card Number')); ?>
```

Comment:
```
var ccerr = <?php echo xjl('Invalid Credit Card Number'); ?>
```

Reply:
if not in 501 don't want to use. i'll json encode.